bin/lib/installation.py

```diff
@@ -773,6 +773,49 @@ def __repr__(self) -> str:
         return f'TarballInstallable({self.name}, {self.install_path})'
 
 
+class ZipArchiveInstallable(Installable):
+    def __init__(self, install_context: InstallationContext, config: Dict[str, Any]):
+        super().__init__(install_context, config)
+        self.install_path = self.config_get('dir')
+        self.url = self.config_get('url')
+        self.folder_to_rename = self.config_get('folder')
+        self.configure_command = command_config(self.config_get('configure_command', []))
+        self.strip = self.config_get('strip', False)
+        self._setup_check_exe(self.install_path)
+
+    def stage(self) -> None:
+        self.install_context.clean_staging()
+        name = self.config_get('name')
+        # Unzip does not support stdin piping so we need to create a file
+        with (self.install_context.staging / 'distribution.zip').open('wb') as fd:
+            self.install_context.fetch_to(self.url, fd)
+            self.install_context.stage_command(['unzip', fd.name])
+            self.install_context.stage_command(['mv', self.folder_to_rename, self.install_path])
```

Comment:
Unlike `tar`, `unzip` does not unzip the files into a folder with the same name as the archive, so a `mv` is used to re name the unzip output folder name to self.install_path

Reply:
Isn't it a matter of how the archive was created? I don't think `tar` adds any automatic prefix (at least not by default)

Reply:
That may be the case, I don't really know too much about how the archives are laid out, but it seems like I get the same kotlinc folder after unzipping, when we would like to have it named self.install_path.

We could use the -d option for unzip which will unzip into a given directory. That means the executable has the path  `f'{self.install_path}/kotlinc/bin/kotlinc-jvm'`.

Reply:
Right; both tar and unzip keep whatever directory structure was placed into the archive. This is why it's configurable on the tar side, as sometimes we need to skip a top level directory (`--strip-components`) or otherwise know what name it expands to vs what we want it to be.

I'm fine to leave it and modify the code as needed when we find something that doesn't work (that's how the rest of the code has grown).